Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,232 @@
+/-
+Copyright (c) 2023 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Analysis.Convex.Combination
+import Mathlib.LinearAlgebra.AffineSpace.Independent
+import Mathlib.Tactic.Linarith
+
+/-!
+# Radon's convexity theorem
+
+Radon's theorem on convex sets states that any affine dependent set can be partitioned into two sets
+whose convex hulls intersect.
+
+## Main results
+
+* `Radon_partition`: Radon convexity theorem
+
+## Tags
+convex hull, radon
+
+-/
+
+
+open Set Finset
+open BigOperators
+
+universe u
+
+variable {𝕜 : Type*} {E : Type u} [LinearOrderedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+
+/-- Any family `f` of affine dependent vectors contains a set `I` with the property that
+convex hulls of `I` and `Iᶜ` intersect. -/
+theorem radon_partition {ι : Type*} {f : ι → E}
+    (h : ¬AffineIndependent 𝕜 f) : ∃ (I : Set ι),
+    (Set.Nonempty ((convexHull 𝕜 (f '' I)) ∩ (convexHull 𝕜 (f '' Iᶜ)))) := by
+  unfold AffineIndependent at h; push_neg at h
```

Comment:
Since you're working in a vector space, you can use [affineIndependent_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/AffineSpace/Independent.html#affineIndependent_iff), which is more convenient that these "weighted VSub".
```suggestion
  rw [affineIndependent_iff] at h; push_neg at h
```